Board.cpp

```diff
@@ -217,37 +236,64 @@ set<int> Board::genMoves() const
 			else // pawn moves
 			{
 				int n, m = (side == white) ? 1 : -1;
-				if (squares[i + m * 8].color == none)
+				uint8_t singlePawnMove = static_cast<uint8_t>(i + m * 8);
+				uint8_t doublePawnMove = static_cast<uint8_t>(i + m * 16);
+				bool pawnPromotionWhite = side == white && (i / 8 == 6 );
+					bool pawnPromotionBlack = side == black && (i / 8 == 1 );
+				if (squares[singlePawnMove].color == none)
 				{
-					move.m = {4, i, static_cast<uint8_t>(i + m * 8), 0};
+					move.m = {4, i, static_cast<uint8_t>(singlePawnMove), 0};
 					moves.insert(move.x);
-					if (side == white && (i / 8 == 1) && squares[i + m * 16].color == none)
+
+					
+                    if(pawnPromotionWhite || pawnPromotionBlack)
+                    {
+						set<int> pawnPromotionMoves = generatePawnPromotionMoves(i, singlePawnMove);
+                        moves.insert(pawnPromotionMoves.begin(), pawnPromotionMoves.end());
+                    }
+                    
+                    //double pawn for white 
+					if (side == white && (i / 8 == 1) && squares[doublePawnMove].color == none)
 					{
-						move.m = {8, i, static_cast<uint8_t>(i + m * 16), 0};
+						move.m = {8, i, doublePawnMove, 0};
 						moves.insert(move.x);
 					}
-					if (side == black && (i / 8 == 6) && squares[i + m * 16].color == none)
+                    //double pawn for black
+					if (side == black && (i / 8 == 6) && squares[doublePawnMove].color == none)
 					{
-						move.m = {8, i, static_cast<uint8_t>(i + m * 16), 0};
+						move.m = {8, i, doublePawnMove, 0};
 						moves.insert(move.x);
 					}
+
 				}
+				//pawn capture square
 				n = mailbox[mailbox64[i] + m * -9];
 				if (n != -1 && squares[n].color == xside)
 				{
 					move.m = {64, i, static_cast<uint8_t>(n), 0};
 					moves.insert(move.x);
+                    if(pawnPromotionWhite || pawnPromotionBlack)
+                    {
+						set<int> pawnPromotionMoves = generatePawnPromotionMoves(i, n);
```

Comment:
fromat

Reply:
Yea we should find something that works for VS code because I am using that too.

Reply:
I'll look up a c++ plugin and then see if we can add a formatter to the code base